### PR TITLE
Upstream support websocket enable and hashOn select

### PR DESCRIPTION
**Pull request by lilien1010**

### CODE REVIEW

File: src/views/schema/upstream/edit.vue [hunk 1]
```diff
@@ -230,7 +288,9 @@ export default class extends Vue {
           type = null,
           nodes = [],
           key = null,
-          desc = ''
+          desc = '',
+          enable_websocket = null,
+          hash_on = null
```
Comment (juzhiyuan) [inline src/views/schema/upstream/edit.vue]: Boolean type

File: src/views/schema/upstream/edit.vue [hunk 2]
```diff
@@ -348,6 +410,13 @@ export default class extends Vue {
       .el-input {
         width: 193px;
       }
+      .tip {
+          line-height: 24px;
```
Comment (juzhiyuan) [inline src/views/schema/upstream/edit.vue]: 2 spaces

File: src/views/schema/upstream/edit.vue [hunk 3]
```diff
@@ -348,6 +410,13 @@ export default class extends Vue {
...
+          font-size: 12px;
+          margin: 0;
+          color: #8e8c8c;
+          display:inline;
```
Comment (juzhiyuan) [inline src/views/schema/upstream/edit.vue]: space

File: src/lang/en.ts [hunk 4]
```diff
@@ -75,5 +75,12 @@ export default {
       inputMultipleValues: 'Input value then press return button',
       propertyHostsTip: 'Any values are supported'
     }
+  },
+  upstream: {
+    websocket: {
+      EnabledTips: 'WebSocket enabled',
+      DisabledTips: 'WebSocket disabled'
+    },
+    keyTips: 'You can edit and input any value here,press Enter to confirm.'
```
[the remark this answers is not in the document]
Comment (lilien1010) [reply src/lang/en.ts]: ok

File: src/views/schema/routes/edit.vue [hunk 5]
```diff
@@ -383,6 +383,7 @@ export default class extends Vue {
           }
         })
 
+        delete data['tempPlugin']
```
[the remark this answers is not in the document]
Comment (lilien1010) [reply src/views/schema/routes/edit.vue]: temp['tempPlugin']==null  will not enter the delete logic so I think we can keep it this way

File: src/views/schema/upstream/edit.vue [hunk 6]
```diff
@@ -53,6 +53,26 @@
         </el-select>
       </el-form-item>
 
+      <el-form-item
+        label="Hash On"
+      >
+        <el-select
+          v-model="form.hash_on"
+          placeholder="Select a hash on"
```
[the remark this answers is not in the document]
Comment (lilien1010) [reply src/views/schema/upstream/edit.vue]: ok

File: src/views/schema/upstream/edit.vue (hunk 2 above)
Comment (lilien1010) [reply src/views/schema/upstream/edit.vue]: ok

**Pull request merged**